[MJAVADOC-437] javadoc:aggregate fails on initial build

hwellmann opened the pull request:
For aggregate reports, do not resolve artifacts of current reactor to make initial build pass.

hwellmann committed:
[MJAVADOC-437] for aggregate reports, do not resolve artifacts of
current reactor to make initial build pass

hwellmann committed:
[MJAVADOC-437] added integration test

michael-o commented on maven-javadoc-plugin/src/it/MJAVADOC-437/module2/pom.xml:
```diff
@@ -0,0 +1,50 @@
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>2.2</version>
+        </dependency>
```
If you need that format properly.

michael-o commented on maven-javadoc-plugin/src/it/MJAVADOC-437/pom.xml:
```diff
@@ -0,0 +1,80 @@
+      <version>3.8.2</version>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
```
Not used in the tests...

michael-o commented on maven-javadoc-plugin/src/it/MJAVADOC-437/pom.xml:
```diff
@@ -0,0 +1,80 @@
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-plugin-plugin</artifactId>
+          <version>2.5</version>
+        </plugin>
```
Not used in the tests...

michael-o commented on maven-javadoc-plugin/src/it/MJAVADOC-437/pom.xml:
```diff
@@ -0,0 +1,80 @@
+                          <configuration>
+                        <source>1.8</source>
+                        <target>1.8</target>
+                    </configuration>
```
There is no need for 1.8, the plugin requires 1.5 max.

michael-o commented on maven-javadoc-plugin/src/it/MJAVADOC-437/pom.xml:
```diff
@@ -0,0 +1,80 @@
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-javadoc-plugin</artifactId>
+      </plugin>
```
Plugin management is fully sufficient, imho.

michael-o commented on maven-javadoc-plugin/src/it/MJAVADOC-437/verify.bsh:
```diff
@@ -0,0 +1,64 @@
+    
+	FileReader fr = new FileReader( javadoc );
+	BufferedReader br = new BufferedReader( fr );
+    String line;
```
The reader has not been closed.

hwellmann committed:
[MJAVADOC-437] test cleanup

removed unneeded dependencies and plugins
fixed formatting
closed BufferedReader
documented the need for an external dependency (e.g. commons-io)

michael-o commented:
@hwellmann This has already been merged into trunk. Please close.
